NBitcoin/Protocol/AddressManager.cs

```diff
@@ -48,7 +48,17 @@ internal class AddressInfo : IBitcoinSerializable
 			public void ReadWrite(BitcoinStream stream)
 			{
 				stream.ReadWrite(ref _Address);
-				stream.ReadWrite(ref source);
+				var useV2Format = stream.ProtocolVersion is {} protcocolVersion && (protcocolVersion &  NetworkAddress.AddrV2Format) != 0;
```

Comment:
IDK what's going on, but related:

[`6f3b967` (#949)](https://github.com/MetacoSA/NBitcoin/pull/949/commits/6f3b96781dc88787b816686afea595df50f98490#diff-8746b2305b8a1420a5604bde479710adc86d056eacaef926159f3686e6d8ffd8R230)

Reply:
Ohh! good catch. I have to revert this change.

Reply:
I mean, something like this will be necessary but this PR doesn't cover the AddressManager serialization, I will address than in a future PR.

Reply:
Done.

Reply:
I have no idea what you did as I didn't understand the code in the first place, but happy that you've found something based on this random note.